ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/SourceBuilder.java

```diff
@@ -508,8 +514,19 @@ public GenericRow transform(final K key, final GenericRow row) {
           final long timestamp = processorContext.timestamp();
           final Collection<?> keyColumns = keyGenerator.apply(key);
 
-          row.ensureAdditionalCapacity(1 + keyColumns.size());
+          final int additionalColumns = SystemColumns
```

Comment:
```suggestion
          final int numPseudoColumns = SystemColumns
```
I think the name `additionalColumns` is confusing because the key columns are "additional" in some sense too.